src/components/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { addTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { User } from '../types/User';
+
+type Props = {
+  todosLength: number
+  setError: (error: string) => void,
+  user: User | null,
+  onAdd: (todo: Todo) => void,
+  isAdding: boolean,
+  setIsAdding: (b: boolean) => void,
+  setNewTodoTitle: (title: string) => void,
+  allCompleted: boolean,
+  onCompleteAll: () => void
+};
+
+export const Header: React.FC<Props> = ({
+  todosLength,
+  setError,
+  user,
+  onAdd,
+  isAdding,
+  setIsAdding,
+  setNewTodoTitle,
+  allCompleted,
+  onCompleteAll,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [titleValue, setTitleValue] = useState('');
+
+  const onSubmitHandler = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!titleValue) {
+      setError('Title can\'t be empty');
+
+      return;
+    }
+
+    if (user) {
+      setNewTodoTitle(titleValue);
+
+      setTitleValue('');
+
+      const newTodo = {
+        id: 0,
+        userId: user.id,
+        title: titleValue,
+        completed: false,
+      };
+
+      try {
+        setIsAdding(true);
+        const response = await addTodo(user.id, newTodo);
+        const {
+          id,
+          userId,
+          title,
+          completed,
+        } = response;
+
+        const todo = {
+          id,
+          userId,
+          title,
+          completed,
+        };
```

Comment:
```suggestion
        const newTodo = await addTodo(user.id, newTodo);
```
won't it work like this?

Reply:
Sure it would. I thought it would be nice to pick only values we need.